src/viam/sdk/components/private/board_client.cpp

```diff
@@ -135,26 +135,17 @@ Board::digital_value BoardClient::read_digital_interrupt(const std::string& digi
 void BoardClient::stream_ticks(std::vector<std::string> const& digital_interrupt_names,
                                std::function<bool(Tick&& tick)> const& tick_handler,
                                const AttributeMap& extra) {
-    viam::component::board::v1::StreamTicksRequest request;
-    viam::component::board::v1::StreamTicksResponse response;
-    ClientContext ctx;
-
-    request.set_name(this->name());
-
-    for (const auto& name : digital_interrupt_names) {
-        request.add_pin_names(name);
-    }
-    *request.mutable_extra() = map_to_struct(extra);
-
-    auto reader = stub_->StreamTicks(ctx, request);
-
-    while (reader->Read(&response)) {
-        if (!tick_handler({response.pin_name(),
-                           std::chrono::nanoseconds(response.time()),
-                           response.high()})) {
-            break;
-        }
-    };
+    return make_client_helper(this, *stub_, &StubType::StreamTicks)
+        .with(extra,
+              [&](auto& request) {
+                  for (const auto& name : digital_interrupt_names) {
+                      request.add_pin_names(name);
+                  }
+              })
+        .invoke_stream([&](auto& response) {
```

Comment:
In light of the findings around `TryCancel` and `Finish`, I think there is now an identified correctness issue with any other streaming client call that isn't making use of those. Can you have a look around to try to see if there are others and just pull them into using the new helper?

Reply:
I think the only ones so far are 
- `Board`, changed in this PR
- `InputController`, forthcoming but based off this PR as well
- `AudioInput`, not yet started
there's also the gizmo example, should I change that too?

I checked for other streaming protos, I think none other are in robot components so they're not covered:
https://github.com/search?q=repo%3Aviamrobotics%2Fapi+path%3A%2F%5Eproto%5C%2F%2F+stream&type=code

Reply:
Great, thanks for checking, I wasn't sure.